x/uregistry/keeper/genesis.go

```diff
@@ -94,7 +95,15 @@ func deployProxyAdminContract(ctx context.Context, evmKeeper types.EVMKeeper, pr
 }
 
 func deploySystemContracts(ctx context.Context, evmKeeper types.EVMKeeper) {
-	for name, contract := range types.SYSTEM_CONTRACTS {
+	// Sort contract names for deterministic iteration order across all validators
+	names := make([]string, 0, len(types.SYSTEM_CONTRACTS))
```

Comment:
This was causing issues in local validators syncing resulting in app hash mismatches.
@0xNilesh to check if the change is fine